### DISCUSSION TITLE
Secondary sort by name/color/score when primary sort is by status

### DISCUSSION THREAD

**Discussion by navinthomas** [iSoron/uhabits#1341] [Feature Requests] [answered]
### Feature
When a user selects sort by status, there should be an additional option to secondary sort by name or color or score. Habits grouped under the same status should be sorted within their respective groups with this secondary sort option.

### My Use Case 
_Context:_ I used to use sort by color earlier. I colored the habits so that when I select sort by color, they appear in a chronological way based on the order in which they should ideally be done. Another way of doing this would be to number the habits by have a number prefix in the habit names. 

I have started using sort by status to have the already completed habits of a day to be at the bottom of the list. That way, the habits that are due for today, and still not done, are shown at the top, in an easily visible manner. Although the habits are grouped under their respective statuses, they unfortunately seem to be sorted in a random order. This randomness is off settling. I would much prefer a secondary sort option like sort by color or name or score, due to the reasons mentioned above. I would still be able to keep habits grouped based on their status, but also have the habits ordered in their chronological order.

**Comment by iSoron** [comment, marked as answer]
Hi @navinthomas, this feature is already available as of Loop 2.0.3, although it's not documented anywhere and it's not discoverable in the UI. If you first sort habits by name, then sort then by color, Loop will use **color** as the primary sort order, and **name** as the secondary one. That is, it will sort habits by color first and, if there are ties, it will break the ties based on names. You can use this with any two sorting methods.

**Reply by navinthomas** [reply]
Great! I saw it happen in the app, but I was not sure if it was just keeping the habits arranged temporarily as per the previous ordering, and whether it will lose this ordering the next day when the habit statuses are reset. Thanks for confirming that it is already available.
